services/graphql/src/mongoose/schema/order.js

```diff
@@ -84,11 +84,9 @@ schema.pre('save', async function updateRelatedModels() {
 schema.pre('save', async function updateEvents() {
   if (this.isModified('advertiserId')) {
     const { advertiserId } = this;
-    ['view', 'click'].forEach((modelName) => {
-      connection.model(modelName).updateMany({ orderId: this._id }, {
-        $set: { advertiserId },
-      }).catch(e => logError(e));
-    });
+    connection.model('event').updateMany({ orderId: this._id }, {
```

Comment:
This query must also include the proper types, e.g:
`{ orderId: this._id, type: { $in: ['view', 'click'] } }`

Reply:
Index this as well